.github/workflows/deploy-dev.yml

```
@@ -0,0 +1,185 @@
+      github.event_name == 'push' ||
+      github.event_name == 'workflow_dispatch'
+    outputs:
+      name: ${{ steps.payu.outputs.name }}
+      version: ${{ steps.payu.outputs.version }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Get Payu Version
+        id: payu
+        run: |
+          # Set version to datetime and last short commit hash of payu
+          NOW=$(date -u +"%Y%m%dT%H%M%SZ")
+          COMMIT_HASH=${{ needs.check-for-payu-updates.outputs.last-commit-hash }}
+          VERSION="dev-$NOW-$COMMIT_HASH"
+          echo "version=$VERSION" >> $GITHUB_OUTPUT
+          echo "name=payu-$VERSION" >> $GITHUB_OUTPUT
+
+      - name: Setup Micromamba
+        uses: mamba-org/setup-micromamba@422500192359a097648154e8db4e39bdb6c6eed7 #v1.8.1
+        with:
+          micromamba-version: '1.5.3-0'
+          environment-file: env-dev.yml
+          environment-name: ${{ steps.payu.outputs.name }}
+          generate-run-shell: true
+      
+      - name: Create Pack
+        shell: micromamba-shell {0}
+        run: conda pack
+
+      - name: Upload Artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: payu-dev
+          if-no-files-found: error
+          path: |
+            ${{ steps.payu.outputs.name }}.tar.gz
+
+  deploy:
+    runs-on: ubuntu-latest
+    needs:
+      - pack
+    environment: Gadi
+    env:
+      NAME: ${{ needs.pack.outputs.name }}
+      VERSION: ${{ needs.pack.outputs.version }}
+    steps:
+      - uses: actions/download-artifact@v3.0.2
+        with:
+          name: payu-dev
+
+      - uses: access-nri/actions/.github/actions/setup-ssh@main
+        id: ssh
+        with:
+          hosts: |
+            ${{ secrets.HOST_DATA }}
+            ${{ secrets.HOST }}
+          private-key: ${{ secrets.SSH_KEY }}
+
+      - name: Copy to Gadi 
+        run: |
+          rsync -e 'ssh -i ${{ steps.ssh.outputs.private-key-path }}' \
+            ${{ env.NAME }}.tar.gz \
+            ${{ secrets.USER }}@${{ secrets.HOST_DATA }}:${{ vars.PRERELEASE_PACK_LOCATION }}
+
+      - name: Deploy to Gadi
+        env:
+          ENVIRONMENT_LOCATION: ${{ vars.PRERELEASE_DEPLOYMENT_LOCATION }}/${{ env.VERSION }}
+          ENVIRONMENT_SYMLINK: ${{ vars.PRERELEASE_DEPLOYMENT_LOCATION }}/dev
+          MODULE_SYMLINK: ${{ vars.PRERELEASE_MODULE_LOCATION }}/dev
+          PACKED_ENVIRONMENT: ${{ vars.PRERELEASE_PACK_LOCATION }}/${{ env.NAME }}.tar.gz
+        run: |
+          ssh ${{ secrets.USER }}@${{ secrets.HOST }} -i ${{ steps.ssh.outputs.private-key-path }} /bin/bash <<'EOT'
+
+          # Create list of previous payu/dev environments to remove later
+          old_versions=$(ls "${{ vars.PRERELEASE_DEPLOYMENT_LOCATION }}" | grep -E '^dev-[0-9]{8}T[0-9]{6}Z-.*')
+        
+          # Unpack conda enviroment
+          mkdir ${{ env.ENVIRONMENT_LOCATION }}
+          if [ $? -ne 0 ]; then
+            exit $?
+          fi
```

Can be simplified to:

```suggestion
          mkdir ${{ env.ENVIRONMENT_LOCATION }} || exit $?     
```

---

Thanks, I've added in that suggestion